elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,348 @@
+
+                // update mutated attributes
+                parentProjection.setAttributes(attributes);
+            }
+        } else {
+            throw new InvalidEntityBodyException(
+                    String.format("Unknown attribute field {%s.%s}.", parentProjection.getName(), fieldName));
+        }
+    }
+
+    /**
+     * Construct Elide {@link Pagination}, {@link Sorting}, {@link Attribute} from GraphQL {@link Argument} and
+     * add it to the {@link EntityProjection}.
+     *
+     * @param argument graphQL argument
+     * @param entityProjection projection that has this argument
+     */
+    private void addArgument(Argument argument, EntityProjection entityProjection) {
+        String argumentName = argument.getName();
+
+        if (isPaginationArgument(argumentName)) {
+            // NOOP
+        } else if (isSortingArgument(argumentName)) {
+            // NOOP
+        } else {
+            addAttributeArgument(argument, entityProjection);
+        }
+    }
+
+    /**
+     * Returns whether or not a GraphQL argument name corresponding to a pagination argument.
+     *
+     * @param argumentName Name key of the GraphQL argument
+     *
+     * @return {@code true} if the name equals to {@link ModelBuilder#ARGUMENT_FIRST} or
+     * {@link ModelBuilder#ARGUMENT_AFTER}
+     */
+    private static boolean isPaginationArgument(String argumentName) {
+        return ModelBuilder.ARGUMENT_FIRST.equals(argumentName) || ModelBuilder.ARGUMENT_AFTER.equals(argumentName);
+    }
+
+
+    /**
+     * Returns whether or not a GraphQL argument name corresponding to a sorting argument.
+     *
+     * @param argumentName Name key of the GraphQL argument
+     *
+     * @return {@code true} if the name equals to {@link ModelBuilder#ARGUMENT_SORT}
+     */
+    private static boolean isSortingArgument(String argumentName) {
+        return ModelBuilder.ARGUMENT_SORT.equals(argumentName);
+    }
+
+    private void addAttributeArgument(Argument argument, EntityProjection entityProjection) {
+        String argumentName = argument.getName();
+        Class<?> entityType = entityProjection.getType();
+
+        if (!entityDictionary.isValidField(entityType, argumentName)) {
+            // invalid argument name
+            throw new IllegalStateException(
+                    String.format("Unknown argument field {%s.%s}.", entityType, argument));
+        }
+
+        Attribute argumentAttribute = entityProjection.getAttributeByName(argumentName);
+        com.yahoo.elide.request.Argument elideArgument = com.yahoo.elide.request.Argument.builder()
+                .name(argumentName)
+                .value(argument.getValue())
+                .build();
+
+        if (argumentAttribute == null) {
+            argumentAttribute = Attribute.builder()
+                    .type(entityDictionary.getType(entityType, argumentName))
+                    .name(argumentName)
```

Comment:
I think this is wrong.  The name of the attribute should be the field name - not the argument name.

Reply:
I think the argument here is something like `id`, which is one of the fields of the projected entity, isn't it?